src/toil/worker.py

```diff
@@ -319,7 +320,7 @@ def recursiveDelete(jobWrapper2):
                     jobWrapper.checkpoint = jobWrapper.command
 
                 # Create a fileStore object for the job
-                fileStore = FileStore(jobStore, jobWrapper, localWorkerTempDir, blockFn)
+                fileStore = FileStore.createFileStore(jobStore, jobWrapper, localWorkerTempDir, blockFn, caching)
```

Comment:
If a Config instance is in scope, it's probably clearer to no extract `caching` and call as `…, caching=not config.disableCaching`.